src/main/java/appeng/core/features/DamagedItemDefinition.java

```diff
@@ -32,7 +34,7 @@
 {
 	private final IStackSrc source;
 
-	public DamagedItemDefinition( IStackSrc source )
+	public DamagedItemDefinition( @Nonnull IStackSrc source )
```

Comment:
I do not think a `Preconditions` is actually needed here. Probably also not for `ItemMultiMaterial.createMaterial()`.

Both are only internal, therefore under our control. The general convention for `assert`/`throw Exception` is usually `assert` for internal code, which does not deal with external input, so it is enough to explicitly activate them for testing (manually, unit tests, etc) and `throw exception` (or `Preconditions`) in case there is an actual need to validate external data.